webapp/src/Service/ImportExportService.php

```diff
@@ -148,9 +148,20 @@ public function importContestData($data, ?string &$message = null, string &$cid
 
         $starttime->setTimezone(new DateTimeZone(date_default_timezone_get()));
         $activateTime = new DateTime();
-        if ($activateTime > $starttime) {
+        if (array_key_exists('activate-time', $data)) {
+            $activatetimeValue = $data['activate-time'];
+            if (is_string($activatetimeValue)) {
+                $activateTime = date_create_from_format(DateTime::ISO8601, $activatetimeValue) ?:
+                    // Make sure ISO 8601 but with the T replaced with a space also works.
+                    date_create_from_format('Y-m-d H:i:sO', $activatetimeValue);
+            } else {
+                /** @var DateTime $activateTime */
+                $activateTime = $activatetimeValue;
+            }
+        } elseif ($activateTime > $starttime) {
             $activateTime = $starttime;
         }
+        $activateTime->setTimezone(new DateTimeZone(date_default_timezone_get()));
```

Comment:
Why this change? This is also changing the behavior of the old case, is that intentional?

Reply:
I've rewrote the PR, so I'll resolve this.